sdk/src/main/java/io/dapr/client/DaprClientBuilder.java

```diff
@@ -97,11 +103,30 @@ public DaprClientBuilder withStateSerializer(DaprObjectSerializer stateSerialize
    * @throws java.lang.IllegalStateException if any required field is missing
    */
   public DaprClient build() {
-    if (this.useGrpc) {
-      return buildDaprClientGrpc();
+    if (this.apiProtocol != this.methodInvocationApiProtocol) {
+      return new DaprClientProxy(buildDaprClient(this.apiProtocol), buildDaprClient(this.methodInvocationApiProtocol));
     }
 
-    return buildDaprClientHttp();
+    return buildDaprClient(this.apiProtocol);
+  }
+
+  /**
+   * Creates an instance of a Dapr Client based on the chosen protocol.
+   *
+   * @param protocol Dapr API's protocol.
+   * @return the GRPC Client.
+   * @throws java.lang.IllegalStateException if either host is missing or if port is missing or a negative number.
+   */
+  private DaprClient buildDaprClient(DaprApiProtocol protocol) {
+    if (protocol == null) {
+      throw new IllegalArgumentException("Protocol is required.");
+    }
+
+    switch (protocol) {
+      case GRPC: return buildDaprClientGrpc();
+      case HTTP: return buildDaprClientHttp();
+      default: throw new IllegalArgumentException("Unsupported protocol: " + protocol.name());
```

Comment:
Why is this exception different and worded differently from the exception in `ActorProxyBuilder`?
There it was:
```java
switch (this.apiProtocol) {
      return new DaprGrpcClient(DaprGrpc.newFutureStub(this.channel));	      case GRPC: return new DaprGrpcClient(DaprGrpc.newFutureStub(this.channel));
      case HTTP: return new DaprHttpClient(daprHttpBuilder.build());
      default: throw new IllegalStateException("Invalid Dapr API protocol.");
}
```

Reply:
OK. Made them the same. IllegalState because it is the state of the builder that does not depend on a param.